pkg/non_kube/common/claim_redeemer.go

```diff
@@ -2,9 +2,121 @@ package common
 
 import (
 	"github.com/skupperproject/skupper/pkg/non_kube/apis"
+
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"time"
+
+	skupperv1alpha1 "github.com/skupperproject/skupper/pkg/apis/skupper/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
 )
 
 func RedeemClaims(siteState *apis.SiteState) error {
-	// TODO Redeem logic that will populate siteState.Secrets and siteState.Links
+	var errs []error
+
+	for name, claim := range siteState.Claims {
+		err := redeemAccessToken(claim, siteState)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to redeem claim %s: %w", name, err))
+			fmt.Printf("RedeemClaims: failed to redeem claim: %s: %s\n", name, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("failed to redeem %d claims", len(errs))
+	}
+
+	return nil
+}
+
+// Redeem logic that populates siteState.Secrets and siteState.Links
+func redeemAccessToken(claim *skupperv1alpha1.AccessToken, siteState *apis.SiteState) error {
+	transport := &http.Transport{}
+	if claim.Spec.Ca != "" {
+		caPool := x509.NewCertPool()
+		caPool.AppendCertsFromPEM([]byte(claim.Spec.Ca))
+		transport.TLSClientConfig = &tls.Config{
+			RootCAs: caPool,
+		}
+	}
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+	request, err := http.NewRequest(http.MethodPost, claim.Spec.Url, bytes.NewReader([]byte(claim.Spec.Code)))
+	request.Header.Add("name", claim.Name)
+	request.Header.Add("subject", string(siteState.Site.ObjectMeta.UID))
```

Comment:
@fgiorgetti I wanted to double-check: is this line correct?  

In my testing, I see that the UID is an empty string.

Should non-kube sites be assigned a UID?  Or, perhaps the `subject` should be populated differently than what's coded here?

Reply:
Updated code to set subject to `Site.Name`.